### PR TITLE
Code review changes

### PR DESCRIPTION
**Pull request by Aaronk41**
Pull Request for code changes

### CODE REVIEW

**Comment by Aaronk41** [comment]
Analysis of the Program:
The DissCardController class has been enhanced with several key improvements aimed at reducing technical debt. These include transitioning to constructor injection for better dependency management, enhancing logging for better traceability, and implementing a utility method for creating ResponseEntity objects. These changes improve the overall readability, maintainability, and robustness of the code, making it easier for future developers to work with.

Timeliness of Program Availability:
Yes, the program was available in the UC GitHub repository on time, allowing for adequate review and collaboration among team members.

Understanding the Program:
As an outsider, the program is relatively easy to understand. The use of clear method names, logging, and structured comments helps to convey the purpose and functionality of the code effectively. The code follows standard conventions, making it accessible to someone familiar with Java and Spring.

Compilation Status:
The program compiles

Commits:
Refactored Card Service]
Improved Logging in DissCardController]
Added Error Handling for Add Card Endpoint]

File: src/main/java/com/teamfive/disscard/DissCardController.java [hunk 1]
```diff
@@ -14,98 +15,101 @@
...
-     * Fetches a card's info based on its ID
-     * @param id id of a card registered with DissCard
-     * @return A JSON object representing a card
+     * Fetches a card's info based on its ID.
+     * @param id ID of a card registered with DissCard.
+     * @return A ResponseEntity containing a JSON object representing a card.
      */
     @GetMapping("/card/info/{id}")
     public ResponseEntity<Card> fetchCardById(@PathVariable("id") int id) {
+        log.info("Fetching card with ID: {}", id);
```
Comment (Aaronk41) [inline src/main/java/com/teamfive/disscard/DissCardController.java:74]: Standardization: The return statement now uses the buildResponse method, ensuring that the response creation logic is consistent across different endpoints. This helps maintain a standardized response format.

File: src/main/java/com/teamfive/disscard/DissCardController.java [hunk 2]
```diff
@@ -14,98 +15,101 @@
...
     public ResponseEntity<Card> addCard(@RequestBody Card card) {
-        Card newCard = null;
-        HttpHeaders headers = new HttpHeaders();
+        log.info("Adding new card: {}", card);
         try {
-            newCard = cardService.save(card);
+            Card newCard = cardService.save(card);
+            return buildResponse(newCard);
+        } catch (Exception e) {
+            log.error("Error adding card: {}", e.getMessage());
```
Comment (Aaronk41) [inline src/main/java/com/teamfive/disscard/DissCardController.java:103]: HTTP Status Code: The status code for a successful addition of a card was changed from HttpStatus.OK to HttpStatus.CREATED (201). This better represents the outcome of the operation, which is the creation of a new resource.

**Review by Aaronk41 — COMMENTED** [review]
review

**Comment by leichtje** [comment]
Cleared all conflicts and merged the branch

**Pull request merged**